dev-libs/libpipeline/libpipeline-1.5.0.recipe

```diff
@@ -0,0 +1,83 @@
+SUMMARY="A C library for setting up and running pipelines"
+DESCRIPTION="libpipeline is a C library for setting up and running pipelines \
+of processes, without needing to involve shell command-line parsing which is \
+often error-prone and insecure.  This alleviates programmers of the need to \
+laboriously construct pipelines using lower-level primitives such as fork(2) \
+and execve(2)."
+HOMEPAGE="http://libpipeline.nongnu.org/"
+COPYRIGHT="1994 Markus Armbruster.
+	1989, 1990, 1991, 1992, 2000, 2001, 2002, 2003, 2004, 2005,
+	2006, 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
+	2003-2017 Colin Watson"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://download.savannah.nongnu.org/releases/libpipeline/libpipeline-$portVersion.tar.gz"
+CHECKSUM_SHA256="0d72e12e4f2afff67fd7b9df0a24d7ba42b5a7c9211ac5b3dcccc5cd8b286f2b"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libpipeline$secondaryArchSuffix = $portVersion
+	lib:libpipeline$secondaryArchSuffix = $portVersion
```

Comment:
```diff
+libVersion=$portVersion
+libVersionCompat="$libVersion compat >= 1.5"
+
 PROVIDES="
 	libpipeline$secondaryArchSuffix = $portVersion
-	lib:libpipeline$secondaryArchSuffix = $portVersion
+	lib:libpipeline$secondaryArchSuffix = $libVersionCompat
	"
```

Reply:
I've seen this passing by with a few others, what does the $libVersionCompat provide? :)

Reply:
On recipes that provide one or more libraries which all have the same soname, adding these 2 custom variables (libVersion and libVersionCompat) can be handy because:
* we only define libVersion once, and we use it in the `lib:lib*` line(s) of PROVIDES and in the  `devel:lib*` line(s) of PROVIDES_devel.
* we define libVersionCompat right after libVersion, with the "compat" stuff, and we can use it in the defineDebugInfoPackage call.
* it simplifies future upgrades: no risk to update the version on a `lib:lib*` line and forget the `devel:lib*` (or the opposite).
* it helps new contributors to understand the difference between the version of the recipe and that of its libraries.

Of course, using these custom variables is just a possibility. I don't expect it to become a rule. It is much better to let each contributor decide if that's a good idea or not.
(Well, I think it is ;-)

Reply:
Ah, you didn't mention "providing libVersion" yet ;) probably why the build failed earlier :)